cpu/cc2538/radio/cc2538_rf.c

```diff
@@ -193,6 +193,11 @@ bool cc2538_on(void)
 
 void cc2538_setup(cc2538_rf_t *dev)
 {
+#if IS_USED(MODULE_IEEE802154_RADIO_HAL)
+    extern ieee802154_dev_t cc2538_rf_dev;
+    netdev_ieee802154_submac_init(&dev->netdev, &cc2538_rf_dev);
+    cc2538_init();
+#else
```

Comment:
You could still do the `netdev_register()` in both cases - as long as we have `netdev_ieee802154_submac`

Reply:
Yes, true. I will try that